src/System.Private.ServiceModel/tests/Common/Scenarios/Endpoints.cs

```diff
@@ -1,49 +1,183 @@
+    private static X509Certificate2 clientCert;
+    private static object rootCertLock = new object();
+    private static object clientCertLock = new object();
+    //Install Root certificate, this is required for all https test
+    private static void EnsureRootCertificateInstalled()
+    {
+        lock (rootCertLock)
+        {
+            if (rootCert == null)
+            {
+                BasicHttpBinding basicHttpBinding = new BasicHttpBinding();
+                ChannelFactory<IUtil> factory = null;
+                IUtil serviceProxy = null;
+                try
+                {
+                    factory = new ChannelFactory<IUtil>(basicHttpBinding, new EndpointAddress(ServiceUtil_Address));
+                    serviceProxy = factory.CreateChannel();
+                    rootCert = new X509Certificate2(serviceProxy.GetRootCert(false));
+
+                    if (rootCert == null)
+                    {
+                        //throw
+                        throw new Exception("Failed to obtain root cert from the server");
+                    }
+
+                    BridgeClientCertificateManager.InstallCertificateToRootStore(rootCert);
+                }
+                finally
+                {
+                    ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+                }
+            }
+        }
+    }
+
+    //Install client certificate
+    private static void EnsureLocalClientCertifciateInstalled()
+    {
+        EnsureRootCertificateInstalled();
+        lock (clientCertLock)
+        {
+            if (clientCert == null)
+            {
+                BasicHttpBinding basicHttpBinding = new BasicHttpBinding();
+                ChannelFactory<IUtil> factory = null;
+                IUtil serviceProxy = null;
+                try
+                {
+                    factory = new ChannelFactory<IUtil>(basicHttpBinding, new EndpointAddress(ServiceUtil_Address));
+                    serviceProxy = factory.CreateChannel();
+                    clientCert = new X509Certificate2(serviceProxy.GetClientCert(false), "test", X509KeyStorageFlags.PersistKeySet);
+                    if (clientCert == null)
+                    {
+                        //throw
+                        throw new Exception("Failed to obtain client cert from the server");
+                    }
+
+                    BridgeClientCertificateManager.AddToStoreIfNeeded(StoreName.My, StoreLocation.CurrentUser, clientCert);
+                    BridgeClientCertificateManager.LocalCertThumbprint = clientCert.Thumbprint;
+                }
+                finally
+                {
+                    ScenarioTestHelpers.CloseCommunicationObjects((ICommunicationObject)serviceProxy, factory);
+                }
+            }
+        }
+    }
+
+    private static string GetServiceHostName()
+    {
+        return TestProperties.GetProperty(TestProperties.BridgeHost_PropertyName);
```

Comment:
+1 about parsing if possible for the tests.  If we need 2 properties, it should no longer be called "BridgeHost"

---
In reply to: [59492624](https://github.com/dotnet/wcf/pull/1002#discussion_r59492624) [](ancestors = 59492624,59492053)